pkg/apis/crd/v1alpha1/types.go

```diff
@@ -354,3 +355,146 @@ type BGPPeer struct {
+	// At the moment this is the only supported configuration and it is required for every capture.
+	FirstN *PacketCaptureFirstNConfig `json:"firstN,omitempty"`
+}
+
+// PacketCaptureFileServer specifies the PacketCapture file server information.
+type PacketCaptureFileServer struct {
+	// The URL of the file server. It is set with format: scheme://host[:port][/path],
+	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
+	URL string `json:"url"`
+}
+
+type PacketCaptureSpec struct {
+	// Timeout is the timeout for this capture session. If not specified, defaults to 60s.
+	Timeout       *uint16       `json:"timeout,omitempty"`
+	CaptureConfig CaptureConfig `json:"captureConfig"`
+	// Source is the traffic source we want to perform capture on. Both `Source` and `Destination` is reuqired
+	// for a capture session, and at least one `Pod` should present either in the source or the destination.
+	Source      Source      `json:"source"`
+	Destination Destination `json:"destination"`
+	// Packet defines what kind of traffic we want to capture between the source and desination. If not spicified,
+	// all kinds of traffic will count.
+	Packet *Packet `json:"packet,omitempty"`
+	// FileServer specifies the sftp url config for a file server. If present, captured packets will be uploaded to this server.
+	// If not, packets file will only be present in the antrea-agent pod.
+	// When capture finished, the path information will be shown in `.status.PacketsFilePath`.
+	FileServer *PacketCaptureFileServer `json:"fileServer,omitempty"`
+}
+
+type PacketCaptureStatus struct {
+	// NumCapturedPackets records how many packets have been captured. If it reaches the target number, the capture
+	// can be considered as finished.
+	NumCapturedPackets int32 `json:"numCapturedPackets,omitempty"`
+	// PacketsFilePath is the file path where the captured packets are stored. The format is: "<antrea-agent-pod-name>:<path>".
+	// If `.spec.FileServer` is present, this file will also be uploaded to the targeted location. This file
+	// will be removed after the PacketCapture CR is deleted.
+	PacketsFilePath string `json:"packetsFilePath"`
+	// StartTime is the time when this capture session starts.
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+	// Condition represents the latest available observations of the PacketCapture 's current state.
+	Conditions []PacketCaptureCondition `json:"conditions"`
+}
+
+type PacketCaptureConditionType string
+
+const (
+
+	// CaptureCompleted means enough packets has been captured and saved in antrea-agent Pod locally already, but hasn't been
+	// uploaded yet(if a fileserver has been configured).
+	CaptureCompleted PacketCaptureConditionType = "CaptureCompleted"
+
+	// PacketsUploaded means the captured packets file has been uploaded to the target fileserver.
+	PacketsUploaded PacketCaptureConditionType = "PacketsUploaded"
```

Comment:
you need a more uniform naming scheme for your conditions. See https://pkg.go.dev/k8s.io/api/core/v1#NodeConditionType

maybe `PacketCaptureCompleted` and `PacketCaptureFileUploaded`

we may need additional conditions based on other discussions to indicate that a capture is pending (because of too many concurrent captures) or that a capture is ongoing but has not completed yet

Reply:
OK, updated.